src/components/LanguageSelector/index.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { publish } from '@edx/frontend-platform';
+import {
+  injectIntl, LOCALE_CHANGED, getLocale, handleRtl, getPrimaryLanguageSubtag,
+} from '@edx/frontend-platform/i18n';
+import { logError } from '@edx/frontend-platform/logging';
+import { Dropdown, useWindowSize } from '@openedx/paragon';
+import { Language } from '@openedx/paragon/icons';
+import { getCookies } from '@edx/frontend-platform/i18n/lib';
+import { patchPreferences, postSetLang } from './data';
+
+const onLanguageSelected = async ({ langCookieName, username, selectedLlocale }) => {
+  try {
+    if (username) {
+      await patchPreferences(username, { prefLang: selectedLlocale });
+      await postSetLang(selectedLlocale);
+    } else {
+      getCookies().set(langCookieName, selectedLlocale);
+    }
+    publish(LOCALE_CHANGED, getLocale());
+    handleRtl();
+  } catch (error) {
+    logError(error);
+  }
+};
+const getLocaleName = (locale) => {
+  const langName = new Intl.DisplayNames([locale], { type: 'language', languageDisplay: 'standard' }).of(locale);
+  return langName.replace(/^\w/, (c) => c.toUpperCase());
```

Comment:
```suggestion
  return langName.charAt(0).toUpperCase() + langName.slice(1);
```
I see that this line capitalizes the string. A more straightforward implementation without regular expressions might be better for this task. I'll leave it up to you.